### PR TITLE
Add APM tracing to Container Orchestrator

### PR DESCRIPTION
**Pull request by jdpgrailsdev**
## What
* Enable APM tracing support for container orchestrator

## How
* Pass required environment variables to enable APM tracing from airbyte-workers to the airbyte-container-orchestrator container launched via Kubernetes
* Add custom tracing to container orchestrator orchestration jobs

## Recommended reading order
1. `ContainerOrchestratorConfigBeanFactory.java`
2. `AsyncOrchestratorPodProcess.java`
3. `NormalizationJobOrchestrator.java`
4. `ReplicationJobOrchestrator.java`
5. `DbtJobOrchestrator.java`

## Tests

* All unit tests pass and platform builds successfully locally
* Platform can be deployed locally via Docker Compose without any issue
* All changes deployed to dev cloud environment with APM enabled.  Verified that traces for container orchestrator appear.

**N.B.** There will be a follow-on cloud PR to actually enable the use of APM in our deployed container orchestrator instance.

### CODE REVIEW

File: airbyte-commons-worker/src/main/java/io/airbyte/workers/process/AsyncOrchestratorPodProcess.java [hunk 1]
```diff
@@ -294,12 +293,9 @@ public void create(final Map<String, String> allLabels,
...
     }
 
-    final EnvConfigs envConfigs = new EnvConfigs();
-    envVars.add(new EnvVar(EnvConfigs.METRIC_CLIENT, envConfigs.getMetricClient(), null));
-    envVars.add(new EnvVar(EnvConfigs.DD_AGENT_HOST, envConfigs.getDDAgentHost(), null));
-    envVars.add(new EnvVar(EnvConfigs.DD_DOGSTATSD_PORT, envConfigs.getDDDogStatsDPort(), null));
-    envVars.add(new EnvVar(EnvConfigs.PUBLISH_METRICS, Boolean.toString(envConfigs.getPublishMetrics()), null));
-    envVars.add(new EnvVar(EnvVariableFeatureFlags.USE_STREAM_CAPABLE_STATE, Boolean.toString(useStreamCapableState), null));
+    // Copy all additionally provided environment variables
+    envVars.addAll(environmentVariables.entrySet().stream().map(e -> new EnvVar(e.getKey(), e.getValue(), null)).collect(Collectors.toList()));
```
Comment (colesnodgrass) [inline airbyte-commons-worker/src/main/java/io/airbyte/workers/process/AsyncOrchestratorPodProcess.java:297]: Can `.collect(Collectors.toList())` be replaced with `.toList()`

File: airbyte-workers/src/main/java/io/airbyte/workers/config/ContainerOrchestratorConfigBeanFactory.java [hunk 2]
```diff
@@ -42,16 +55,41 @@ public ContainerOrchestratorConfig kubernetesContainerOrchestratorConfig(
...
+    environmentVariables.put(EnvVariableFeatureFlags.USE_STREAM_CAPABLE_STATE, Boolean.toString(featureFlags.useStreamCapableState()));
+    environmentVariables.put(JAVA_OPTS_ENV_VAR, containerOrchestratorJavaOpts);
+
+    if (System.getenv(DD_ENV_ENV_VAR) != null) {
+      environmentVariables.put(DD_ENV_ENV_VAR, System.getenv(DD_ENV_ENV_VAR));
+    }
+
+    if (System.getenv(DD_VERSION_ENV_VAR) != null) {
+      environmentVariables.put(DD_VERSION_ENV_VAR, System.getenv(DD_VERSION_ENV_VAR));
+    }
```
Comment (colesnodgrass) [inline airbyte-workers/src/main/java/io/airbyte/workers/config/ContainerOrchestratorConfigBeanFactory.java:87]: Should this be added to the `application.yml` file?

**Review by colesnodgrass — APPROVED** [review]

File: airbyte-commons-worker/src/main/java/io/airbyte/workers/process/AsyncOrchestratorPodProcess.java (hunk 1 above)
Comment (jdpgrailsdev) [reply airbyte-commons-worker/src/main/java/io/airbyte/workers/process/AsyncOrchestratorPodProcess.java:297]: Yes...will do.

File: airbyte-workers/src/main/java/io/airbyte/workers/config/ContainerOrchestratorConfigBeanFactory.java (hunk 2 above)
Comment (jdpgrailsdev) [reply airbyte-workers/src/main/java/io/airbyte/workers/config/ContainerOrchestratorConfigBeanFactory.java:87]: What is going on here is that the env vars are set in the Helm chart, as they need to be available for the DD Java agent to pick up.   We are then copying the ones set on the worker for use in the orchestrator process that will be launched.  We could also use injection but we still have to set it as an env var, so I felt it was just best to leave it as a direct read of the env var.

**Review by benmoriceau — APPROVED** [review]

**Pull request merged**